docs/docs/guides/integrations/mcp.md

```diff
@@ -0,0 +1,236 @@
+
+## Installation
+
+To use the MCP integration with Weave, you'll need to install both weave and the MCP package:
+
+```bash
+pip install -qq mcp[cli] weave
+```
+
+## Quickstart Guide
+
+### Prerequisites
+
+Make sure you have both Weave and MCP installed as described in the Installation section.
+
+### Using the Example Code
+
+Weave includes ready-to-use MCP examples. Instead of creating files from scratch, you can use the examples provided in the Weave repository:
+
+1. Clone the Weave repository if you haven't already:
+   ```bash
+   git clone https://github.com/wandb/weave
+   cd weave
+   ```
+
+2. Install the required dependencies:
+   ```bash
+   pip install -e ".[mcp]"
+   ```
+
+3. Navigate to the MCP example directory:
+   ```bash
+   cd examples/mcp_demo
+   ```
+
+### Running the Example
+
+The example consists of two main files:
+- `example_server.py`: A demo MCP server with various tools, resources, and prompts
+- `example_client.py`: A client that connects to the server
+
+To run the example:
+
+```bash
+python example_client.py example_server.py
+```
+
+This will start the client, which will connect to and interact with the server. The client provides a command-line interface with the following options:
+
+- `tools` - List all available tools
+- `resources` - List all available resources
+- `prompts` - List all available prompts
+- `add <a> <b>` - Add two numbers
+- `bmi <weight_kg> <height_m>` - Calculate BMI
+- `weather <city>` - Get weather for a city
+- `greeting <name>` - Get personalized greeting
+- `user <id>` - Get user profile
+- `config` - Get application configuration
+- `code-review <code>` - Get code review prompt
+- `debug <error>` - Get debug error prompt
+- `demo` - Run demos for all features
+- `q` - Exit the session
+
+If you type `demo`, the client will run through all the features. Typing `q` will close the process. If you want to play with the available features, try individual commands listed above.
+
+You will find a trace timeline as shown below:
+
+[![mcp_trace_timeline.png](imgs/mcp/mcp_trace_timeline.png)](https://wandb.ai/ayut/mcp_example/weave/traces?filter=%7B%22opVersionRefs%22%3A%5B%22weave%3A%2F%2F%2Fayut%2Fmcp_example%2Fop%2Frun_client%3A*%22%5D%7D&peekPath=%2Fayut%2Fmcp_example%2Fcalls%2F01966bbe-cc5e-7012-b45f-bf10617d8c1e%3FhideTraceTree%3D0)
+
+By default, you will just see the `run_client` traces. Click on the Ops selection box and select "All Calls"
+
+[!mcp_all_calls.png](imgs/mcp/mcp_all_calls.png)]
+
+Doing so will show you `FastMCP` methods (tools, resources, prompts) traced by the integration. You can see the arguments given to the tools and returned values.
+
+[!mcp_fastmcp.png](imgs/mcp/mcp_fastmcp.png)](https://wandb.ai/ayut/mcp_example/weave/traces?peekPath=%2Fayut%2Fmcp_example%2Fcalls%2F01966bc2-aca1-7021-a626-aecfe677b1b4%3FhideTraceTree%3D0)
```

Comment:
this image should be more zoomed in on the properties of the trace, its too zoomed out to notice. optionally add a box to highigt where the reader should be looking